content/blog/2019/secure-egress-control-istio/index.md

```diff
@@ -0,0 +1,323 @@
+
+In all the cases you have to apply some
+[additional security mechanisms](/docs/examples/advanced-gateways/egress-gateway/#additional-security-considerations),
+like Kubernetes Network Policies or an L3 firewall that will enforce that traffic from the cluster to the outside is
+allowed for the egress gateway only. See
+[an example for Kubernetes Network Policies configuration](/docs/examples/advanced-gateways/egress-gateway/#apply-kubernetes-network-policies).
+
+You must also increase the security measures applied to Istio control plane and the
+egress gateway by running them on nodes separate from the application nodes, in a separate namespace, monitoring them
+more thoroughly, etc. After all, if the attackers are able to attack Istio Mixer or the egress gateway, they could
+violate any policy.
+
+Once you direct egress traffic through an egress gateway and apply the additional security mechanisms,
+you can securely monitor the traffic and define security policies on it.
+If the application sends HTTP requests and the egress gateway performs TLS origination, you can monitor HTTP
+information, e.g HTTP methods, headers and URL paths, and you can
+[define policies](/blog/2018/egress-monitoring-access-control) based on the HTTP information. If the application
+performs TLS origination, for TLS traffic you can
+[monitor SNI and the service account](/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/) of the
+source pod, and define policies based on them.
+
+The following diagram shows Istio Security Architecture, augmented with L3 Firewalls (part of the
+[additional security mechanisms](/docs/examples/advanced-gateways/egress-gateway/#additional-security-considerations)
+provided outside of Istio by the cluster/cloud provider).
+The L3 Firewalls can have a trivial configuration that would allow incoming traffic into Istio ingress gateway pods only and
+outgoing traffic from Istio egress gateway pods only. Note that the Istio proxy of the egress gateway performs
+policy enforcement and reporting in the same way as the sidecar proxies in the application pods.
+
+{{< image width="80%" link="./SecurityArchitectureWithL3Firewalls.svg" caption="Istio Security Architecture with Egress Gateway and L3 Firewalls" >}}
+
+Now let's examine possible attacks and let me show you how the secure egress control in Istio prevents them.
+
+### Possible attacks and their prevention
+
+Consider the following security policy with regard to egress traffic:
+
+1. Application A is allowed to access `*.ibm.com` (all the external services with URL matching `*.ibm.com`,
+  e.g. `www.ibm.com`)
+1. Application B is allowed to access `mongo1.composedb.com`
+1. All the egress traffic must be monitored
+
+Now consider a scenario in which the application A (one of the pods) is compromised. Suppose the attackers have the
```

Comment:
in which one of application A's pods is compromised.

Reply:
Fixed in https://github.com/istio/istio.io/pull/3179/commits/1831dcb73de8bed6a0736b27594b4c180512ccdb